edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -86,9 +88,40 @@ internal static CreateContainerParameters GetCreateOptions(string createOptions)
         }
 
         internal static string ValidateAndGetImage(string image)
+        {
+            return ValidateAndGetImage(image, new EnvironmentWrapper());
+        }
+
+        internal static string ValidateAndGetImage(string image, IEnvironmentWrapper env)
         {
             image = Preconditions.CheckNonWhiteSpace(image, nameof(image)).Trim();
+
+            if (image[0] == '$')
```

Comment:
>if (image[0] == '$') [](start = 12, length = 20)

isn't this check already part of your regex expression?

Reply:
It allows flagging if user wanted to put $upstream but instead made a typo: $upsteam

Reply:
if the user put $ then we assume they want to use an $upstream. We then check for sanity and create exception if wrong.